cmd/vela-email/plugin.go

```diff
@@ -0,0 +1,282 @@
+// Copyright (c) 2021 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package main
+
+import (
+	"bytes"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"html/template"
+	"net/smtp"
+	"os"
+	"strings"
+
+	"github.com/aymerick/douceur/inliner"
+	"github.com/jordan-wright/email"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	// ErrorMissingEmailToParam is returned when the plugin is missing the To email parameter.
+	ErrorMissingEmailToParam = errors.New("missing email parameter: To")
+
+	// ErrorMissingEmailFromParam is returned when the plugin is missing the From email parameter.
+	ErrorMissingEmailFromParam = errors.New("missing email parameter: From")
+
+	// ErrorEmptyAttach is returned when the plugin finds the provided attachment to be empty.
+	ErrorEmptyAttach = errors.New("attachment provided is empty")
+
+	// ErrorMissingSMTPParam is returned when the plugin is missing a smtp host or port parameter.
+	ErrorMissingSMTPParam = errors.New("missing smtp parameter (host/port)")
+
+	// ErrorMissingSMTPUsernameParam is returned when the plugin is missing smtp username parameter.
+	ErrorMissingSMTPUsernameParam = errors.New("missing smtp username")
+
+	// ErrorMissingSMTPPasswordParam is returned when the plugin is missing smtp password parameter.
+	ErrorMissingSMTPPasswordParam = errors.New("missing smtp password")
+)
+
+// Plugin represents the configuration loaded for the plugin.
+type (
+	Plugin struct {
+		// Email arguments loaded for the plugin
+		Email *email.Email
+		// Attachment arguments loaded for the plugin
+		Attachment *email.Attachment
+		// SmtpHost arguments loaded for the plugin
+		SMTPHost *SMTPHost
+		// TLSConfig arguments loaded for the plugin
+		TLSConfig *tls.Config
+		// SendType arguments loaded for the plugin
+		SendType string
+		// Auth arguments loaded for the plugin
+		Auth string
+		// Env arguments loaded for the plugin
+		Env map[string]string
+		// Readable build time environment variables
+		BuildEnv *BuildEnv
+	}
+
+	// SMTPHost struct.
+	SMTPHost struct {
+		Host     string
+		Port     string
+		Username string
+		Password string
+	}
+
+	// User friendly readable Build Environment Variables.
+	BuildEnv struct {
+		BuildCreated  string
+		BuildEnqueued string
+		BuildFinished string
+		BuildStarted  string
+	}
+)
+
+// Validate checks the plugin parameters needed from the
+// user are provided. If the email subject or HTML/text are
+// not provided, defaults are set.
+func (p *Plugin) Validate() error {
+	logrus.Trace("entered plugin.Validate")
+	defer logrus.Trace("exited plugin.Validate")
+
+	logrus.Info("Validating Parameters...")
+	if len(p.Attachment.Filename) != 0 {
+		fileInfo, err := os.Stat(p.Attachment.Filename)
+		if errors.Is(err, os.ErrNotExist) {
+			return os.ErrNotExist
+		}
+
+		if fileInfo.Size() == 0 {
+			return ErrorEmptyAttach
+		}
+
+		// Don't check email's errors
```

Comment:
what do these comments mean?
```
		// Don't check email's errors
```

Reply:
They can be removed, it means just return the error that the email code returns, don't double check that they returned the right error and don't create a new one